nixos/modules/services/security/clamav-unofficial-sigs.nix

```diff
@@ -0,0 +1,87 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.clamav-unofficial-sigs;
+in {
+  options.services.clamav-unofficial-sigs = with types; {
+    enable = mkEnableOption "Unofficial ClamAV signatures";
+
+    user = mkOption {
+      type = str;
+      description = "User to run systemd service under";
+      default = "clamav";
+    };
+
+    group = mkOption {
+      type = str;
+      description = "Group to run systemd service under. Should be the same as used by clamav in order to allow this service access to the clamav database.";
+      default = "clamav";
+    };
+
+    extraConfig = mkOption {
+      type = attrsOf str;
+      description = "Extra key-value pairs to configure in user.conf";
+      example = { malwarepatrol_receipt_code = "YOUR-RECEIPT-NUMBER"; };
+      default = {};
+    };
+
+    extraConfigFiles = mkOption {
```

Comment:
Why files and not more config values?

Reply:
Some options contain credentials of paid services and we use a shared configuration repository. Therefore we place credentials directly in files on the target host and include them, so there are no secrets in the nix config repository.